### DISCUSSION TITLE
How to reset loading state for specific model/effect?

### DISCUSSION THREAD

**Discussion by vaytsel** [rematch/rematch#978] [Q&A]
How to reset loading state for specific model/effect?

**Comment by semoal** [comment]
Loading it's a state like the traditional ones. You can perfectly `dispatch.loading.modelName.effectName.hide();` or `dispatch.loading.modelName.effectName.show()`;

**Reply by vaytsel** [reply]
thank u!

**Comment by vaytsel** [comment]
@semoal Hello again! Today I tried your solution and it didn't work(

I have loading state like:

```javascript
{ 
 loading: {
    global: {success: false, error: false ...},
    models: {anyModel: {...}},
    effects: {anyModel: { fetchSmth: { loading: false ... } } }
 }
}
```

When I try call `dispatch.loading.effects.anyModel.fetchSmth.show()`, I catch an exception `Uncaught TypeError: Cannot read properties of undefined (reading 'anyModel')`.

With `dispatch.loading.anyModel.fetchSmth.hide();` same problem - `Uncaught TypeError: Cannot read properties of undefined (reading 'fetchSmth')`.

What am I doing wrong?

**Comment by vaytsel** [comment]
I find way through 

```javascript
export const store = init<RootModel, FullModel>({
    models,
    plugins: [selectPlugin(), loadingPlugin({ type: 'full' })],
    redux: {
        rootReducers: {
            RESET_APP: () => undefined,
        },
        devtoolOptions: {
            disabled: process.env.NODE_ENV === 'production',
        },
    },
});

dispatch({ type: 'RESET_APP' })
```
this solution reseting the all state... but I need reset specific effect loading state

**Comment by vaytsel** [comment]
@semoal Hello, could you recheck my question please?

**Reply by juanpicado** [reply]
I also had to use this solution https://github.com/rematch/rematch/issues/663#issuecomment-527119154 which does the job but I also could not make it work, I could use  `dispatch.loading.hide()` which is valid but does nothing, Icc: @semoal 
<img width="469" alt="Screenshot 2022-11-11 at 11 19 52" src="https://user-images.githubusercontent.com/558752/201345885-462d96a8-9daf-459f-a8a3-c6c6e6554ee1.png">

**Reply by vaytsel** [reply]
@juanpicado I find way, use like this

`dispatch({ type: 'loading/hide', payload: { name: 'modelName', action: 'effectName' } });`

**Reply by juanpicado** [reply]
Awesome thanks for sharing !!
